src/logging.rs

```diff
@@ -112,40 +124,39 @@ pub mod trace {
         /// do not exercise the `proxy-log-level` endpoint.
         pub fn dangling() -> Self {
             let builder = subscriber_builder()
-                .with_filter(filter::EnvFilter::default())
+                .with_filter(Filter::default())
                 .with_filter_reloading();
             let inner = builder.reload_handle();
-            LevelHandle { inner }
+            LevelHandle { inner, current: Arc::new(Mutex::new(String::new())) }
         }
 
         pub fn set_level(&self, level: impl AsRef<str>) -> Result<(), Error> {
             let level = level.as_ref();
-            let filter = level.parse::<filter::EnvFilter>()?;
+            let filter = level.parse::<Filter>()?;
             self.inner.reload(filter)?;
+            let mut current = self.current.lock().unwrap();
```

Comment:
Just curious why `set_level` has to take a `&self` instead of a `&mut self`.

Reply:
We _could_ take `&mut self`, but we would still need to use locks internally. This is because admin service is `Clone`d, so we need to share one "current" state across all instances of this type. Therefore, taking `&mut self` isn't necessary, since we never actually mutate `self`.